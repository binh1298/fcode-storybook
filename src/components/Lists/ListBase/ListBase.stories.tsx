import { Box as MaterialBox } from "@material-ui/core";
import {
    InboxRounded as InboxIcon,
    StarBorderRounded as StarIcon,
    SendOutlined as SendIcon,
    DeleteOutline as DeleteIcon,
    ReportOutlined as ReportIcon,
    DraftsRounded as DraftsIcon,
} from "@material-ui/icons";

import ListBase, { ListBaseProps } from ".";
import ListItemIconBase from "../../ListItemIcons/ListItemIconBase";
import ListItemTextBase from "../../ListItemTexts/ListItemTextBase";
import ListItemBase from "../../ListItems/ListItemBase";

import { Story } from "@storybook/react";

export default {
    title: "Components/ListBase",
    subcomponents: { ListItemBase, ListItemIconBase, ListItemTextBase },
    component: ListBase,
};

export const SimpleList: Story<ListBaseProps> = (args) => (
    <ListBase {...args}>
        <MaterialBox color="secondary.main" bgcolor="primary.main">
            <ListItemBase>
                <MaterialBox ml={3}>
                    <ListItemIconBase>
                        <MaterialBox color="secondary.main" bgcolor="primary.main">
                            <InboxIcon />
                        </MaterialBox>
                    </ListItemIconBase>
                </MaterialBox>
                <ListItemTextBase
                    primary={
                        <MaterialBox color="secondary.main" fontWeight="fontWeightBold">
                            Inbox
                        </MaterialBox>
                    }
                />
            </ListItemBase>
        </MaterialBox>
        <ListItemBase>
            <MaterialBox ml={3}>
                <ListItemIconBase>
                    <MaterialBox color="primary.contrastText" bgcolor="primary.main">
                        <StarIcon />
                    </MaterialBox>
                </ListItemIconBase>
            </MaterialBox>
            <ListItemTextBase primary="Starred" />
        </ListItemBase>

        <ListItemBase>
            <MaterialBox ml={3}>
                <ListItemIconBase>
                    <MaterialBox color="primary.contrastText" bgcolor="primary.main">
                        <SendIcon />
                    </MaterialBox>
                </ListItemIconBase>
            </MaterialBox>
            <ListItemTextBase primary="Sent" />
        </ListItemBase>

        <ListItemBase>
            <MaterialBox ml={3}>
                <ListItemIconBase>
                    <MaterialBox color="primary.contrastText" bgcolor="primary.main">
                        <DeleteIcon />
                    </MaterialBox>
                </ListItemIconBase>
            </MaterialBox>
            <ListItemTextBase primary="Trash" />
        </ListItemBase>

        <ListItemBase>
            <MaterialBox ml={3}>
                <ListItemIconBase>
                    <MaterialBox color="primary.contrastText" bgcolor="primary.main">
                        <ReportIcon />
                    </MaterialBox>
                </ListItemIconBase>
            </MaterialBox>
            <ListItemTextBase primary="Spam" />
        </ListItemBase>

        <ListItemBase>
            <MaterialBox ml={3}>
                <ListItemIconBase>
                    <MaterialBox color="primary.contrastText" bgcolor="primary.main">
                        <DraftsIcon />
                    </MaterialBox>
                </ListItemIconBase>
            </MaterialBox>
            <ListItemTextBase primary="Drafts" />
        </ListItemBase>
    </ListBase>
);
