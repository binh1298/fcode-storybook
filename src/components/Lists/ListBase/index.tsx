import { List as MaterialList, ListProps as MaterialListProps } from "@mui/material";
import { styled } from "@mui/material/styles";

export interface ListBaseProps extends MaterialListProps {}

const List = styled(MaterialList)<ListBaseProps>(({ theme }) => ({
    backgroundColor: theme.palette.primary.main,
    paddingLeft: "1rem",
    paddingRight: "1rem",
}));

const ListBase = (props: ListBaseProps) => {
    return <List {...props} />;
};

export default ListBase;
