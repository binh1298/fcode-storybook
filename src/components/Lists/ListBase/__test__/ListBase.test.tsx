import ListBase, { ListBaseProps } from "..";

import { render, screen } from "@testing-library/react";

describe("<ListBase />", () => {
    let props: ListBaseProps;

    it("should exist children with default style", () => {
        props = {
            children: "Test",
        };
        render(<ListBase {...props} />);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist children with customized width", () => {
        props = {
            children: "Test",
            disablePadding: true,
        };
        render(<ListBase {...props} />);
        expect(screen.getByText("Test")).toBeInTheDocument();
    });
});
