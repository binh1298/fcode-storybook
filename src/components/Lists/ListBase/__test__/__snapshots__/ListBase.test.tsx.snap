// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListBase /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <ul
      class="MuiList-root MuiList-padding"
    >
      <div>
        Test
      </div>
    </ul>
  </div>
</div>
`;
